File: src/main/java/com/mintos/weather/entity/WeatherEntity.java
package com.mintos.weather.entity;

import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;
import java.util.UUID;

@Table("WEATHER")
public record WeatherEntity(
    UUID id,
    String condition,
    BigDecimal temperatureCelsius,
    BigDecimal windMph,
    String windDirection,
    Long lastUpdatedEpoch,
    UUID ipRegionId
) {
}

File: src/main/kotlin/com/mintos/weather/entity/WeatherEntity.kt
package com.mintos.weather.entity

import org.springframework.data.relational.core.mapping.Table
import java.math.BigDecimal
import java.util.UUID

@Table("WEATHER")
data class WeatherEntity(
    val id: UUID,
    val condition: String,
    val temperatureCelsius: BigDecimal,
    val windMph: BigDecimal,
    val windDirection: String,
    val lastUpdatedEpoch: Long,
    val ipRegionId: UUID
)